### PR TITLE
SDAP-249 : add solr time out client argument to webapp, use it in the data acces…

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
…s configuration

As described in jira ticket SDAP-249

### CODE REVIEW

File: analysis/webservice/algorithms/doms/DomsInitialization.py [hunk 1]
```diff
@@ -49,6 +49,7 @@ def init(self, config):
         log.info("Cassandra Protocol Version: %s" % (cassVersion))
 
         dc_policy = DCAwareRoundRobinPolicy(cassDatacenter)
+        #dc_policy = WhiteListRoundRobinPolicy([cassHost])
```
Comment (eamonford) [inline analysis/webservice/algorithms/doms/DomsInitialization.py:52]: Could you delete this commented line and the unused import in this file?

File: analysis/webservice/webapp.py [hunk 2]
```diff
@@ -156,6 +155,23 @@ def async_callback(self, result):
         if hasattr(result, 'cleanup'):
             result.cleanup()
 
+def inject_args_in_config(args, config):
+    """
+        Takes command argparse arguments and push them in the config
+         with syntax args.<section>_<option>
+    """
+    for _, t_opt in args._options.items():
```
Comment (eamonford) [inline analysis/webservice/webapp.py:163]: you can change this line to `for t_opt in args._options.values():`

File: analysis/webservice/webapp.py [hunk 3]
```diff
@@ -156,6 +155,23 @@ def async_callback(self, result):
         if hasattr(result, 'cleanup'):
             result.cleanup()
 
+def inject_args_in_config(args, config):
```
Comment (eamonford) [inline analysis/webservice/webapp.py:158]: Can you write a unit test for this function? There is some parsing logic which should be unit tested

File: data-access/nexustiles/dao/CassandraProxy.py [hunk 4]
```diff
@@ -168,6 +168,7 @@ def __init__(self, config):
     def __open(self):
 
         dc_policy = DCAwareRoundRobinPolicy(self.__cass_local_DC)
+        #dc_policy = WhiteListRoundRobinPolicy([self.__cass_url ])
```
Comment (eamonford) [inline data-access/nexustiles/dao/CassandraProxy.py:171]: Could you delete this commented line and the unused import in this file?

File: data-access/nexustiles/dao/SolrProxy.py [hunk 5]
```diff
@@ -36,12 +36,18 @@ class SolrProxy(object):
...
+            solr_kargs["timeout"] = config.get("solr", "time_out")
         self.logger = logging.getLogger('nexus')
 
         with SOLR_CON_LOCK:
             solrcon = getattr(thread_local, 'solrcon', None)
             if solrcon is None:
-                solrcon = pysolr.Solr('http://%s/solr/%s' % (self.solrUrl, self.solrCore))
+                solr_url = 'http://%s/solr/%s' % (self.solrUrl, self.solrCore)
+                self.logger.info("connect to solr, url {}".format(solr_url))
+                self.logger.info("with option(s) = {}".format(solr_kargs))
```
Comment (eamonford) [inline data-access/nexustiles/dao/SolrProxy.py:49]: Could you combine these two log statements into one? The NEXUS logs are kind of polluted and these two logs are part of the same sentence anyway

File: analysis/webservice/algorithms/doms/DomsInitialization.py (hunk 1 above)
Comment (tloubrieu-jpl) [reply analysis/webservice/algorithms/doms/DomsInitialization.py:52]: ok, I did that.

File: analysis/webservice/webapp.py (hunk 2 above)
Comment (tloubrieu-jpl) [reply analysis/webservice/webapp.py:163]: Ok, done

File: analysis/webservice/webapp.py (hunk 3 above)
Comment (tloubrieu-jpl) [reply analysis/webservice/webapp.py:158]: I wrote some unit test. I did some with mock before with java, I made my first mock in python.
I can not say I had more fun with it 😄

File: data-access/nexustiles/dao/CassandraProxy.py (hunk 4 above)
Comment (tloubrieu-jpl) [reply data-access/nexustiles/dao/CassandraProxy.py:171]: Done

File: data-access/nexustiles/dao/SolrProxy.py (hunk 5 above)
Comment (tloubrieu-jpl) [reply data-access/nexustiles/dao/SolrProxy.py:49]: Done

File: analysis/webservice/webapp.py [hunk 6]
```diff
@@ -25,10 +25,16 @@
 import pkg_resources
 import tornado.web
 from tornado.options import define, options, parse_command_line
-
 from webservice import NexusHandler
 from webservice.webmodel import NexusRequestObject, NexusProcessingException
 
+logging.basicConfig(
```
Comment (fgreg) [inline analysis/webservice/webapp.py:31]: Care should be taken when configuring `logging` in globally like this. Now, whenever this module is imported, logging will be configured (unless it was already configured by some other module which also does this). Generally, it is better to configure logging in `__main__`, is there a reason you wanted to move it?

Comment (tloubrieu-jpl) [reply analysis/webservice/webapp.py:31]: Ok thanks Franck, I moved back this section in __main__
The reason why I moved it initially is that I wanted to use the log outside the __main__. I created a local one instead.

**Pull request merged**